addons/bus/controllers/main.py

```diff
@@ -38,3 +38,7 @@ def poll(self, channels, last, options=None):
         if request.registry.in_test_mode():
             raise exceptions.UserError(_("bus.Bus not available in test mode"))
         return self._poll(request.db, channels, last, options)
+
+    @route('/longpolling/im_status', type="json")
+    def im_status(self, partner_ids):
+        return dispatch.im_status(request.db, partner_ids)
```

Comment:
Why isn't this simply the following one-liner (without sudo!)?
```suggestion
        return request.env['res.partner'].search_read([['id', 'in', partner_ids]], ['id', 'im_status'])
```